src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Analyzer.java

```diff
@@ -366,7 +366,7 @@ abstract class RedundantLocalVarTypeAnalyzerBase<X extends JCStatement> extends
         }
 
         boolean isImplicitlyTyped(JCVariableDecl decl) {
-            return decl.vartype.pos == Position.NOPOS;
+            return decl.declaredUsingVar();
```

Comment:
an argument of an implicit lambda is implicitly typed but not declared using `var`, I guess we should at least add a note here

Reply:
In fact, this method `isImplicitlyTyped()` is misleadingly named - it appears to be simply testing for whether the variable is declared using `var` (and so maybe it should have been named `isDeclaredUsingVar()`).

But now that `JCVariableDecl.declaredUsingVar()` exists, we can just use that method directly and get rid of the intermediate method, which is not adding any value.

See if 151d9882bef looks better - thanks.